src/auth/client/mod.rs

```diff
@@ -78,48 +80,159 @@ pub enum LoginResponse {
 		redirect_address: (LuString33, u16),
 		/// Whether the account is in free trial mode.
 		is_ftp: bool,
+		/// Whether the account is connecting as a paid account for the first time.
+		just_upgraded_from_ftp: bool,
+		language: Language,
+		country_code: LuString3,
+		auth_logs: Vec<LogEntry>,
 	} = 1,
 	/// The login failed in an unusual way. More information can be found in the attached message.
 	CustomMessage(LuVarWString<u16>) = 5,
 	/// Username or password was incorrect.
 	InvalidUsernamePassword = 6,
 }
 
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct LogEntry {
```

Comment:
I forgot to mention this, but the client calls these "stamps", would be good to keep that name for consistency.